app/model/repository/Comments.php

```diff
@@ -31,4 +32,57 @@ public function persistComment(Comment $comment) {
   public function persistThread(CommentThread $thread) {
     $this->persist($thread);
   }
+
+  /**
+   * Get the number of visible comments for given user
+   * @param $threadId Id of the comments thread.
+   * @param $userId Id of the viewer or null (if nobody is logged in).
+   * @param bool $allVisible True = all coments visible for the user, false = comments made by the user.
+   */
+  public function getThreadCommentsCount(CommentThread $thread, ?User $user, bool $allVisible = true)
+  {
+    if (!$user) {
+      return 0; // anonymous users cannot see any comments
+    }
+
+    $qb = $this->comments->createQueryBuilder('tc')
+      ->select('COUNT(tc.id)')
+      ->where('tc.commentThread = :id');
+
+    if ($allVisible) {
+      $qb->andWhere('tc.user = :user OR tc.isPrivate = 0');
```

Comment:
It might be better to filter this based on ACL - the logic of what the isPrivate flag means should not be repeated here. This might lead e.g. to the comment being hidden from a superadmin. However, since this method only fetches the count of comments, I'm willing to turn my head.

Reply:
The whole point is NOT to fetch all entities and filter them using ACL, especially if we need the count.

Reply:
I understand. But still, how about adding a comment that the method doesn't use the ACL?